typegraph/core/src/utils/fs_host.rs

```diff
@@ -214,3 +216,64 @@ pub fn hash_file(path: &Path) -> Result<(String, u32), String> {
 pub fn path_exists(path: &Path) -> Result<bool, String> {
     path_exists_host(&path.to_string_lossy())
 }
+
+pub fn is_glob(path: &str) -> bool {
+    let path = PathBuf::from(path);
+    if let Some(base_name) = path.file_name() {
+        let base_name = base_name.to_str().unwrap();
+        if base_name.contains('*') || base_name.contains('?') {
+            return true;
+        }
+    }
+
+    false
+}
+
+pub fn get_dir(path: &str) -> PathBuf {
+    let directory = PathBuf::from(path);
+
+    PathBuf::from(directory.parent().unwrap())
+}
+
+pub fn get_matching_files(glob_pattern: &str) -> Result<Vec<PathBuf>, String> {
+    let mut matching_files = vec![];
```

Comment:
Not a prioarity but maybe rename it this to `expand_glob` ? `get_matching_files` is sort of too abstract if I didn't read the arg name to get more context.